### PR TITLE
Velocity penalty drag test

### PR DESCRIPTION
**Pull request by nicholasmalaya**
Now that I am actually using this feature, let's add at least a regression checking that it has not changed in GRINS. Not sure anyone else will use this, but given that this is getting added to the thesis and likely a publication, it is worth regressing. Should have minimal impact on GRINS 'make check' runtime, anyway. 

Test is passing on my local ICES box, Magus.

I've tried to document the essence of it in the input, but it is admittedly pretty scant. Will want to add some model documentation describing what it is doing to the GRINS document, eventually.

### CODE REVIEW

File: test/input_files/velocity_penalty_drag.in [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+[]
+
+[BoundaryConditions]
+   bc_ids = '1:3 0:2'
+   bc_id_name_map = 'Flowing Walls'
+
+   [./Flowing]
+      [./Velocity]
+         type = 'parsed_dirichlet'
+	 # u = '10*(abs(y-.5)<.25)*(y-.25)*(.75-y)'
```
Comment (pbauman) [inline test/input_files/velocity_penalty_drag.in]: Any reason to keep this comment?

File: test/input_files/velocity_penalty_drag.in [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+# quadratic scaling term gives discontinuous second derivatives and
+# kills our numerical jacobian convergence
+verify_analytic_jacobians = 1.e-4
+numerical_jacobian_h = 1.e-10
+
+# Visualization options
+[vis-options]
+output_vis = false
+vis_output_file_prefix = convection_cell
+output_format = 'ExodusII xdr'
```
Comment (pbauman) [inline test/input_files/velocity_penalty_drag.in]: Please take out the ExodusII for the test.

**Comment by pbauman** [comment]
Just a couple of small nitpicks and then this is good. Thanks much!

File: test/input_files/velocity_penalty_drag.in (hunk 1 above)
Comment (nicholasmalaya) [inline test/input_files/velocity_penalty_drag.in]: Absolutely not! Good eyes, I made a pass to clean all the hacking out, but I missed one here.

**Comment by nicholasmalaya** [comment]
Thanks for checking this out. I just pushed commits with both edits (I believe).

**Pull request merged**